node/tools/src/config.rs

```diff
@@ -152,13 +152,19 @@ impl ProtoFmt for AppConfig {
 
 /// This struct holds the file path to each of the config files.
 #[derive(Debug)]
-pub struct ConfigPaths<'a> {
+pub struct ConfigArgs<'a> {
+    /// Node configuration from command line.
+    pub config: Option<AppConfig>,
     /// Path to a JSON file with node configuration.
-    pub app: &'a Path,
+    pub config_file: &'a Path,
+    /// Validator key as a string.
+    pub validator_key: Option<String>,
     /// Path to a validator key file.
-    pub validator_key: Option<&'a Path>,
+    pub validator_key_file: &'a Path,
```

Comment:
this field is now mutually exclusive with validator_key, no? Make it an enum.

Reply:
or actually you might be better off defining a separate ConfigArgs IN ADDITION to ConfigPath. They don't seem to have much overlap.

Reply:
Not sure I understand your comments here. I made them mutually exclusive [here](https://github.com/matter-labs/era-consensus/blob/b63f987057fce56731cc7ebd6098a651fbb88930/node/tools/src/main.rs#L54)

In any case I intend to keep previous behaviour: If no `--validator_key`, nor `--validator_key_file` are defined, still assume `./validator_key` as default if that file is present in root dir. 

If not, then run in non-validator mode

Reply:
If they are mutually exclusive on the CLI arg level, you are currently losing that restriction here and that what makes `ConfigArgs::load` more complicated.

Reply:
I'm not following here. What restriction I'm losing?

To be fair, this functionality had a bug previously: if no `--validator-key` was defined, nor `./validator_key` file was present, the executable raised an error instead of running in non-validator mode.

`ConfigArgs::load` got a bit more complicated, that's true, but I'm not sure how to improve it.

Reply:
> I'm not following here. What restriction I'm losing?

The restriction of mutual exclusivity, and so the ability to represent it as as enum, instead of 2 separate fields which can both exist. 
However, because default values are used as well, it might be less applicable.

Reply:
I mean, can't you just make "specifying config via command line arg" and "specifying config via file(s)" mutually exclusive as a whole? We don't have a use case for combining those, or do we?

Reply:
Combining them is the normal use-case for actual users, where static config files are the "default", and the nested CLI args are the dynamic override. 
However, this is not mandatory, especially when the node standalone use-case is test-only.

Reply:
I was pretty sure that we have the configuration using CLI args only because of kubernetes/docker limitations, i.e. that it is impossible to run a pre-built docker image with access to a dynamically constructed file.

Reply:
Now I made this set of options mutually exclusive: 
You'll need full cli configuration or full file configuration (if no configuration args are provided at all, default files are expected)
Also, if no `--validator-key` or `--validator-key-file` is provided, nor a `validator-key` file exists in root, it won't complain but it should run in non-validator mode